deps/v8/src/sampler.cc

```diff
@@ -342,6 +342,14 @@ void SignalHandler::HandleProfilerSignal(int signal, siginfo_t* info,
     // We require a fully initialized and entered isolate.
     return;
   }
+
+  // ISSUE-14576: To avoid deadlock, return if there is a thread lock
+  if (Isolate::GetProcessWideMutexPointer()->TryLock()) {
```

Comment:
@misterdjules This was instead of, rather than in addition to the pthread_mutex_trylock() approach. Blocking was not added inside the signal handler, but rather around the code in the rest of the system that called pthread_mutex_lock() and pthread_mutex_unlock() (on the process_wide_mutex_) so that when the signal occurs it it will never interrupt those calls to ...lock()/..unlock().